lib/honeycomb/client.rb

```diff
@@ -44,11 +47,35 @@ def initialize(configuration:)
       end
     end
 
-    def start_span(name:, serialized_trace: nil, **fields)
+    def propagation_context_from_req(env:)
+      custom_hook = @additional_trace_options[:parser_hook]
```

Comment:
This logic still seems fiddly - Why are we not just setting a default custom hook to a honeycomb implementation, which just calls these two lines below? I remember discussing this, but don't remember if we came up with a reason to not do it. Also, it really feels like `custom_hook` should be `parser_hook`.

Reply:
You're totally right, and I got caught up with all the trickiness in passing functions around. With the modules as they were before, which you can't really set as a namespaced default value so I gave up on trying to do that. Now that I'm using a class this should be more straightforward, I'll give it a whirl.

Reply:
@dsoo Ok! So it looks like the proper way to store a method in a variable is to use `lambda`. This way we can store that default, overwrite it with the custom hook if it exists, and then just call that variable as a method. We obviously still need the "is there a config?" logic somewhere but I think this is more straightforward and in line with what you're describing? Let me know what you think!